folly/system/ThreadId.h

```diff
@@ -82,6 +82,8 @@ inline uint64_t getOSThreadID() {
   uint64_t tid;
   pthread_threadid_np(nullptr, &tid);
   return tid;
+#elif __has_include(<pthread_np.h>)
```

Comment:
As a general rule, we should avoid using `__has_include` to guard anything but (1) includes and (2) defining owned and namespaced symbols to `1`.

E.g.

```
#if __has_include(<foo>)
#include <foo>
#endif
```

```
#if __has_include(<foo>)
#define MYLIB_HAS_FOO 1
#else
#define MYLIB_HAS_FOO 0
#endif
```

Reply:
That makes sense. Looking for prior art, it looks like LLVM itself struggled with this dance of ifdefs here: 
https://reviews.llvm.org/rG757dbc9ff325eaccaf2bfba8414c205d21117d1c

I've rearranged it based on their gating of __linux__ for the syscall case.